src/client/src/containers/RightSidebar/index.tsx

```diff
@@ -104,6 +106,31 @@ class RightSidebar extends React.Component<Props, IRightSidebarState> {
     selectWebApp: () => {},
     selectPages: () => {}
   };
+
+  state: IRightSidebarState = {
+    isSidebarOpen: false,
+    isSidebarUserControlled: false
+  };
+
+  static getDerivedStateFromProps(
```

Comment:
Does this function use anywhere?

Reply:
I think this was initially added in case we wanted to open up the right side bar on certain pages (ex. azure page and pages page) instead of defaulting it to close

Reply:
Currently it is not being used, but we can make the right side bar display in another pr which would use this function

Reply:
isn't this a lifecycle method? or is this method not being called at all?

Reply:
@piotrmark do you want to comment?

Reply:
never mind is called whenever the component is rendered

Reply:
Yes, It's a lifecycle method, rather important one :D
It opens sidebar when user adds a page or Azure service